src/main/scala/io/citrine/lolo/trees/multitask/MultiTaskTrainingNode.scala

```diff
@@ -147,17 +151,20 @@ class MultiTaskTrainingNode(
       case (_, Some(theRightChild)) if right.nonEmpty =>
         theRightChild.getNode(index)
       case (_, _) =>
+        // TODO (PLA-10388): get the rng in here somehow (right now it's instantiating a random rng)
```

Comment:
This isn't an issue in practice because the typical base learner, `GuessTheMeanLearner`, trains deterministically. It needs to be fixed before the next release, but I'd like to do it in a follow-up PR.

Reply:
What interface needs to be adjusted to allow this?

Reply:
At a high level I think that the training node classes need to be rethought of as being like Learners. Even if they don't inherit explicitly from the `Learner` trait, we should think about them as having a configuration to which we pass data and then when queried they perform logic to calculate the split and produce the model node. Right now the split calculation happens upon instantiation and the model node creation happens later when `getNode` is called. That separation is, I think, fundamentally why we can't thread the `rng` through properly.

Because the "configuration" and "training" steps here are inherently intertwined I think in practice it makes more sense to combine them. Instead of having `getNode` be a function that performs logic, the `node` will be a value in the `TrainingNode` case class. This means `TrainingNode` can't be instantiated directly. Instead it will have a builder method and _this method will take the rng_. It will use that rng to calculate the split, which has an element of randomness. It will then call the builder methods of the left and right nodes (if they exist) and pass the rng to those methods. Then it will return the node, which is now a case class that knows everything already and does no calculation.

This is half-baked but it's what I'll try first.

tl;dr: all of the `Node`s will have their constructors change and they will get builder methods.